### DISCUSSION TITLE
umijs4 请求服务端产生状态码非200时，表单层loading关闭问题。

### DISCUSSION THREAD

**Discussion by leshalv** [umijs/umi#10009] [General]
umijs4 请求服务端产生状态码非200时，会进入异常处理器，上层表单有 loading 开启状态，发生异常后loading状态一直开启，有没有优雅的方式进行关闭，现在是在每个提交中通过try catch 进行处理的。

**Comment by fz6m** [comment]
是的，请求一般都是要 try catch ，可以把 `getApi` 请求包一个含有 try catch 的 hook 在用。

另外推荐使用 [react-query](https://github.com/TanStack/query) 或 [swr](https://github.com/vercel/swr) 来处理 loading 态，可以避免竟态、自带 loading 态，乐观更新等。
